sklearn/covariance/_graph_lasso.py

```diff
@@ -485,20 +543,26 @@ def fit(self, X, y=None):
         # Covariance does not make sense for a single feature
         X = self._validate_data(X, ensure_min_features=2, ensure_min_samples=2)
 
-        if self.assume_centered:
+        if self.covariance:
```

Comment:
```suggestion
        if self.covariance == "precomputed":
```

Reply:
Actually, I wonder if it should be named `precomputed` and be a bool. What do you think @glemaitre ?

Reply:
It is inspired by `SpectralClustering` and the `affinity` parameter. We could think that in the future, we could accept a callable that computes the covariance from `X`. Having `precomputed=True/False` will lock us in.

Here, we just did not want to introduce new API changes but it might be safer for future development.